src/main/java/me/karakelley/http/UriQueryParser.java

```diff
@@ -0,0 +1,45 @@
+package me.karakelley.http;
+
+import me.karakelley.http.exceptions.InvalidRequestException;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URI;
+import java.net.URLDecoder;
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class UriQueryParser {
+
+  public static Map<String, List<String>> parse(String requestPath) {
+    String queries = getQueryString(requestPath);
+    if (queries == null || queries.isEmpty()) {
+      return Collections.emptyMap();
+    }
+
+    return Arrays.stream(queries.split("&"))
+            .map(query -> query.split("="))
+            .collect(Collectors.toMap(pair -> decode(parsePair(pair, 0)),
+                    pair -> Arrays.asList(decode(parsePair(pair, 1)))));
+
+  }
+
+  private static String getQueryString(String requestPath) {
+    try {
+      return URI.create(requestPath).getRawQuery();
+    } catch (Exception e) {
+      throw new InvalidRequestException("Invalid");
```

Comment:
1. Are you catching Exception because of `URI.create(requestPath)` or `.getRawQuery()`? If the former, I'm wondering if there's a way you could verify the requestPath is valid ahead of time so that this can focus exclusively on the queryParams?
2. If there's a way to bomb out somewhere else, it'd be nice if this could return an empty string so that you can avoid the `if (queries == null || queries.isEmpty())` check in `parse`. (I'm pretty sure you could just pass `""` to `Arrays.stream` with all the subsequent chained methods and it'll just be a no-op... but maybe I'm wrong about that?)

Reply:
The reason I throw the exception there (or bomb out) is if the query string has malformed Syntax then a `UriSyntaxException` is thrown. But I realized I can get rid of the `isEmpty()` check since `getRawQuery()` returns `null` vs an empty string.